src/rule_config.jl

```diff
@@ -14,3 +14,38 @@ function ChainRulesCore.rrule_via_ad(config::ADviaRuleConfig, f, args...; kws...
     ret === nothing && throw(MethodError(rrule, (f, args...)))
     return ret
 end
+
+# For testing this config uses finite differences to evaluate the frule and rrule
+Base.@kwdef struct ADviaFDConfig <: RuleConfig{Union{HasReverseMode, HasForwardsMode}}
+    fdm = FiniteDifferences.central_fdm(5, 1)
+end
+
+function ChainRulesCore.frule_via_ad(config::ADviaFDConfig, ȧrgs, f, args...; kws...)
+
+    call_on_copy(f, xs...) = deepcopy(f)(deepcopy(xs)...; deepcopy(kws)...)
+
+    primals_and_tangents = auto_primal_and_tangent.((f, args...))
+    primals = primal.(primals_and_tangents)
+    is_ignored = isa.(ȧrgs, NoTangent)
+
+    Ω = call_on_copy(f, args...)
+    ΔΩ = _make_jvp_call(config.fdm, call_on_copy, Ω, primals, ȧrgs, is_ignored)
+
+    return Ω, ΔΩ
+end
+
+function ChainRulesCore.rrule_via_ad(config::ADviaFDConfig, f, args...; kws...)
+
+    call(f, xs...) = f(xs...; kws...)
+
+    primals_and_tangents = auto_primal_and_tangent.((f, args...))
+    primals = primal.(primals_and_tangents)
+    accum_cotangents = tangent.(primals_and_tangents)
+    is_ignored = isa.(accum_cotangents, NoTangent)
```

Comment:
Do we need to fill in `is_ignored`?
Or would it be ok to return invalid derivatives for these ones.
I suspect the answer is that it would not be OK.
But I wish it wasn't.

I think this section should have a comment explaining that the reason we are calling `auto_primal_and_tangent` is just to try and workout which things we should set `is_ignored` for.

Reply:
I think we do need it, because otherwise we get dimension mismatch in `_make_j′vp_call`

Added a comment that explains the block